_posts/2021-10-25-g2p-applications.md

```diff
@@ -0,0 +1,79 @@
+
+We take a model for doing forced alignment on English and we manually write a `g2p` mapping from the writing system to IPA for the language we want to make a ReadAlong for. For example's sake let's say, Kanyen'kéha (Mohawk). This step requires somebody who is both familiar with the writing system used, and with the International Phonetic Alphabet. 
+
+Then, we use the very cool Python library [PanPhon](https://github.com/dmort27/panphon) to figure out the mapping between the IPA characters in Mohawk and their *closest* IPA equivalents in English. We call this mapping a "Kanyen'kéha IPA to English IPA" mapping. If you have already created your mapping between the orthography and the IPA, you can generate this "Kanyen'kéha IPA to English IPA" using the `g2p` command line. 
+
+Then, we convert the ReadAlong in question from its original orthographic form all the way to English IPA (well, actually to 'Arpabet' which is an ASCII-compliant phonetic transcription standard that the alignment model understands).
+
+Then, after we "do the alignment" (ie figure out which parts of the audio correspond to which part of the text), ReadAlongs puts it all back together again and ta-da! There is your aligned audio and text! Keep reading to see how to generate an X-to-English IPA mapping for your language.
+
+#### Generate your mapping between your language's IPA and English IPA
+
+After [installing]({{ "g2p-basic-mappings-local" | absolute_url }}) `g2p` and once you have made your mapping, update `g2p` by running `g2p update`. When you add or modify any mapping, `g2p` doesn't actually see it yet. `g2p update` is what scans all the mappings and makes them available to `g2p`. Then, you can run `g2p generate-mapping <input-code> --ipa`. So in our example for Kanyen'kéha (moh) we would run 
+
+```bash
+g2p generate-mapping moh --ipa
+```
+
+That will automatically generate a moh-ipa to eng-ipa mapping in `g2p/mappings/langs/generated` - do not edit this file or its configuration unless you really know what you're doing, because it will get overwritten. After generation, you can run `g2p update` again and your mapping will be [usable]({{ "g2p-basic-mappings-local" | absolute_url }}) within `g2p`!
+
+#### Visualizing the process
+
+This is actually where the introductory picture to this blog post comes from! In the graphic below, you can see the word 'bonjour' in French converted to its IPA transcription. Then, its IPA transcription gets converted into its corresponding *English* IPA transcription based on the generated mapping (notice how uvular /ʁ/ gets transformed to alveolo-palatal /ʒ/). Then finally, the English IPA form is converted to Arpabet. So in all, we have an input of 'bonjour' that gets output as 'B AO N ZH UW ZH' and the aligner runs on that form.
```

Comment:
"to this blog post" -> "to this blog series", since the picture is in blog post 1.